test/Interop/Cxx/objc-correctness/const-ref-parameter.swift

```diff
@@ -0,0 +1,77 @@
+// REQUIRES: objc_interop
+
+import ConstRefParameter
+
+func testFunction() {
+    let a = OptionsStruct(intOption: 1, floatOption: 2.0)
+
+    let objc = OptionsConsumerObjC(options: a)
+    _ = objc.doOtherThing(withOptions: a)
+    _ = OptionsConsumerObjC.consumer(withOptions: a)
+    _ = OptionsConsumerObjC.doThing(withOptions: a)
+
+    var cxx = OptionsConsumerCxx(a)
+    _ = cxx.doOtherThing(a)
+    _ = OptionsConsumerCxx.build(a)
+    _ = OptionsConsumerCxx.doThing(a)
+}
+
+// RUN: %target-swift-ide-test -print-module -module-to-print=ConstRefParameter -I %S/Inputs -source-filename=x -enable-experimental-cxx-interop -enable-objc-interop | %FileCheck -check-prefix=CHECK-IDE-TEST %s
+
+// CHECK-IDE-TEST: class OptionsConsumerObjC
+// CHECK-IDE-TEST: init(options: OptionsStruct)
+// COM: FIXME: should it be consumer(options:)?
+// CHECK-IDE-TEST: class func consumer(withOptions options: OptionsStruct) -> Self
+// COM: FIXME: should it be doThing(options:)?
+// CHECK-IDE-TEST: class func doThing(withOptions options: OptionsStruct) -> Int32
+// COM: FIXME: should it be doOtherThing(options:)?
+// CHECK-IDE-TEST: func doOtherThing(withOptions options: OptionsStruct) -> Float
+
+// CHECK-IDE-TEST: struct OptionsConsumerCxx
+// CHECK-IDE-TEST: init(_ options: OptionsStruct)
+// CHECK-IDE-TEST: static func build(_ options: OptionsStruct) -> OptionsConsumerCxx
+// CHECK-IDE-TEST: static func doThing(_ options: OptionsStruct) -> Int32
+// CHECK-IDE-TEST: mutating func doOtherThing(_ options: OptionsStruct) -> Float
+
+
+// RUN: %target-swift-frontend -c -enable-experimental-cxx-interop -enable-objc-interop -I %S/Inputs %s -emit-sil -o - | %FileCheck %s
```

Comment:
Nit: let's check silgen to match all the other tests. Some early sil passes might move things around.

Reply:
For clarity, I mean `-emit-silgen`.